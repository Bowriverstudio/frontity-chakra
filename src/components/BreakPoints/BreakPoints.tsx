import { Badge, useBreakpointValue } from "@chakra-ui/react";
import React from "react";

/**
 * BreakPoints
 */
const BreakPoints = (): JSX.Element => {
  const breakPointsLabels = useBreakpointValue({
    base: "base",
    sm: "sm",
    md: "md",
    lg: "lg",
    xl: "xl",
  });

  const breakPointsColors = useBreakpointValue({
    base: "green",
    sm: "red",
    md: "teal",
    lg: "blue",
    xl: "orange",
  });

  return (
    <>
      <Badge p={4} colorScheme={breakPointsColors}>
        {breakPointsLabels}
      </Badge>
    </>
  );
};

export default BreakPoints;
