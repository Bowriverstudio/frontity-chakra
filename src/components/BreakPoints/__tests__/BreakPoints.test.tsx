/**
 * React / Frontity dependencies
 */
import { ChakraProvider } from "@chakra-ui/react";
import React from "react";
import ReactDOM from "react-dom";
import TestRenderer from "react-test-renderer";

/**
 * Internal dependencies
 */
import BreakPoints from "../BreakPoints";

describe("Frontity Chakra Ui - Breakpoints", () => {
  beforeAll(() => {
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // Deprecated
        removeListener: jest.fn(), // Deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  test("Breakpoints", () => {
    const json = TestRenderer.create(
      <ChakraProvider>
        <BreakPoints />
      </ChakraProvider>
    ).toJSON();
    expect(json).toMatchSnapshot();
  });

  test("Breakpoints renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <ChakraProvider>
        <BreakPoints />
      </ChakraProvider>,
      div
    );
    expect(ReactDOM.unmountComponentAtNode(div)).toBe(true);
  });
});
