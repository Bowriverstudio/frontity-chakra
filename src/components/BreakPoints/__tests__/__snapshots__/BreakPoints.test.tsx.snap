// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Frontity Chakra Ui - Breakpoints Breakpoints 1`] = `
<span
  className="chakra-badge css-19z746w"
/>
`;
